File: LabsSimonaBurlacu232/Lab3/ChessGame/ChessGame/Observers/KillCounterObserver.cs
using System;

namespace ChessGame.Observers
{
    public class KillCounterObserver : IObserver
    {
        private int counter;
        private IObservable observable;

        public KillCounterObserver(IObservable observable)
        {
            counter = 0;
            this.observable = observable;
        }

        public void Subscribe()
        {
            observable.Attach(this);

        }

        public void Unsubscribe()
        {
            observable.Dettach(this);
        }

        public void Update()
        {
            counter++;
            Console.WriteLine("This is the attack number "+counter);
        }
    }
}

File: Lab3Chess/src/KillCounterObserver.java
import pieces.ChessPiece;

public class KillCounterObserver implements MyObserver {
    private int killCount = 0;

    @Override
    public void update(ChessPiece cp, String action) {
        if(action == "attack"){
            killCount++;
            System.out.println("KILL COUNTER: "+killCount);
        }
    }
}
